lib/spack/spack/compiler.py

```diff
@@ -293,14 +289,42 @@ def determine_implicit_rpaths(cls, paths):
             output = str(compiler_exe(cls.verbose_flag(), fin, '-o', fout,
                                       output=str, error=str))  # str for py2
 
-            return cls.parse_implicit_rpaths(output)
+            return cls._parse_non_system_link_dirs(output)
         except spack.util.executable.ProcessError as pe:
             tty.debug('ProcessError: Command exited with non-zero status: ' +
                       pe.long_message)
             return []
         finally:
             shutil.rmtree(tmpdir, ignore_errors=True)
 
+    @classmethod
+    def verbose_flag(cls):
+        """
+        This property should be overridden in the compiler subclass if a
+        verbose flag is available.
+
+        If it is not overridden, it is assumed to not be supported.
+        """
+
+    @classmethod
+    def _parse_non_system_link_dirs(cls, string):
```

Comment:
This should move outside of the class into a module method, since it doesn't make use of the cls variable anywhere.